### ISSUE TITLE
test_ghostwriter_example_outputs[base64_magic]: test_roundtrip_b32hexencode_b32hexdecode missing from output

### ISSUE THREAD

**Issue by hroncok** [HypothesisWorks/hypothesis#3036]
With hypothesis 6.14.3, Python 3.10.0b4 and the following packages installed:

```
Package                       Version
----------------------------- --------
alabaster                     0.7.12
appdirs                       1.4.4
asgiref                       3.3.4
attrs                         21.2.0
Babel                         2.9.1
black                         21.6b0
Bottleneck                    1.3.2
chardet                       4.0.0
click                         8.0.1
colorama                      0.4.4
commonmark                    0.9.1
cycler                        0.10.0
Django                        3.2.1
docutils                      0.16
execnet                       1.9.0
future                        0.18.2
idna                          2.10
imagesize                     1.2.0
iniconfig                     1.1.1
Jinja2                        3.0.1
kiwisolver                    1.3.1
lark-parser                   0.9.0
MarkupSafe                    2.0.0
matplotlib                    3.4.2
mypy-extensions               0.4.3
numexpr                       2.7.1
numpy                         1.20.1
olefile                       0.46
packaging                     21.0
pandas                        1.2.4
pathspec                      0.8.1
pexpect                       4.8.0
Pillow                        8.3.1
pip                           21.1.3
pluggy                        0.13.1
ptyprocess                    0.6.0
py                            1.10.0
pycairo                       1.20.0
Pygments                      2.9.0
pyparsing                     2.4.7
PySocks                       1.7.1
pytest                        6.2.4
pytest-forked                 1.3.0
pytest-xdist                  2.3.0
python-dateutil               2.8.1
pytz                          2021.1
redis                         3.5.3
regex                         2021.7.6
requests                      2.25.1
rich                          10.6.0
scipy                         1.7.0
setuptools                    57.1.0
six                           1.16.0
snowballstemmer               1.9.0
sortedcontainers              2.4.0
Sphinx                        4.0.2
sphinx-hoverxref              0.5b1
sphinx-rtd-theme              0.5.2
sphinx-selective-exclude      1.0.3
sphinxcontrib-applehelp       1.0.2
sphinxcontrib-devhelp         1.0.2
sphinxcontrib-htmlhelp        2.0.0
sphinxcontrib-jsmath          1.0.1
sphinxcontrib-qthelp          1.0.3
sphinxcontrib-serializinghtml 1.1.5
sqlparse                      0.4.1
tables                        3.6.1
toml                          0.10.2
typing-extensions             3.7.4.3
urllib3                       1.26.6
wheel                         0.36.2
```

We see the following test failure on Fedora 35 RPM package build:

```
$ /usr/bin/pytest -vvvv -n auto -k 'not test_registered_from_entrypoint' --ignore tests/codemods --ignore tests/dpcontracts --ignore tests/redis
...
________________ test_ghostwriter_example_outputs[base64_magic] ________________
[gw1] linux -- Python 3.10.0 /usr/bin/python3
Traceback (most recent call last):
  File "/builddir/build/BUILD/hypothesis-hypothesis-python-6.14.3/hypothesis-python/tests/ghostwriter/test_expected_output.py", line 159, in test_ghostwriter_example_outputs
    assert actual == expected  # We got the expected source code
AssertionError: assert ('# This test code was written by the `hypothesis.extra.ghostwriter` module\n'\n '# and is provided under the Creative Commons Zero public domain dedication.\n'\n '\n'\n 'import base64\n'\n 'from hypothesis import given, strategies as st\n'\n '\n'\n '# TODO: replace st.nothing() with appropriate strategies\n'\n '\n'\n '\n'\n '@given(\n'\n '    adobe=st.booleans(),\n'\n '    b=st.nothing(),\n'\n '    foldspaces=st.booleans(),\n'\n '    ignorechars=st.just(b" \\t\\n\\r\\x0b"),\n'\n '    pad=st.booleans(),\n'\n '    wrapcol=st.just(0),\n'\n ')\n'\n 'def test_roundtrip_a85encode_a85decode(adobe, b, foldspaces, ignorechars, '\n 'pad, wrapcol):\n'\n '    value0 = base64.a85encode(\n'\n '        b=b, foldspaces=foldspaces, wrapcol=wrapcol, pad=pad, adobe=adobe\n'\n '    )\n'\n '    value1 = base64.a85decode(\n'\n '        b=value0, foldspaces=foldspaces, adobe=adobe, '\n 'ignorechars=ignorechars\n'\n '    )\n'\n '    assert b == value1, (b, value1)\n'\n '\n'\n '\n'\n '@given(casefold=st.booleans(), s=st.nothing())\n'\n 'def test_roundtrip_b16encode_b16decode(casefold, s):\n'\n '    value0 = base64.b16encode(s=s)\n'\n '    value1 = base64.b16decode(s=value0, casefold=casefold)\n'\n '    assert s == value1, (s, value1)\n'\n '\n'\n '\n'\n '@given(casefold=st.booleans(), map01=st.none(), s=st.nothing())\n'\n 'def test_roundtrip_b32encode_b32decode(casefold, map01, s):\n'\n '    value0 = base64.b32encode(s=s)\n'\n '    value1 = base64.b32decode(s=value0, casefold=casefold, map01=map01)\n'\n '    assert s == value1, (s, value1)\n'\n '\n'\n '\n'\n '@given(casefold=st.booleans(), s=st.nothing())\n'\n 'def test_roundtrip_b32hexencode_b32hexdecode(casefold, s):\n'\n '    value0 = base64.b32hexencode(s=s)\n'\n '    value1 = base64.b32hexdecode(s=value0, casefold=casefold)\n'\n '    assert s == value1, (s, value1)\n'\n '\n'\n '\n'\n '@given(altchars=st.none(), s=st.nothing(), validate=st.booleans())\n'\n 'def test_roundtrip_b64encode_b64decode(altchars, s, validate):\n'\n '    value0 = base64.b64encode(s=s, altchars=altchars)\n'\n '    value1 = base64.b64decode(s=value0, altchars=altchars, '\n 'validate=validate)\n'\n '    assert s == value1, (s, value1)\n'\n '\n'\n '\n'\n '@given(b=st.nothing(), pad=st.booleans())\n'\n 'def test_roundtrip_b85encode_b85decode(b, pad):\n'\n '    value0 = base64.b85encode(b=b, pad=pad)\n'\n '    value1 = base64.b85decode(b=value0)\n'\n '    assert b == value1, (b, value1)\n'\n '\n'\n '\n'\n '@given(input=st.nothing(), output=st.nothing())\n'\n 'def test_roundtrip_encode_decode(input, output):\n'\n '    value0 = base64.encode(input=input, output=output)\n'\n '    value1 = base64.decode(input=value0, output=output)\n'\n '    assert input == value1, (input, value1)\n'\n '\n'\n '\n'\n '@given(s=st.nothing())\n'\n 'def test_roundtrip_encodebytes_decodebytes(s):\n'\n '    value0 = base64.encodebytes(s=s)\n'\n '    value1 = base64.decodebytes(s=value0)\n'\n '    assert s == value1, (s, value1)\n'\n '\n'\n '\n'\n '@given(s=st.nothing())\n'\n 'def test_roundtrip_standard_b64encode_standard_b64decode(s):\n'\n '    value0 = base64.standard_b64encode(s=s)\n'\n '    value1 = base64.standard_b64decode(s=value0)\n'\n '    assert s == value1, (s, value1)\n'\n '\n'\n '\n'\n '@given(s=st.nothing())\n'\n 'def test_roundtrip_urlsafe_b64encode_urlsafe_b64decode(s):\n'\n '    value0 = base64.urlsafe_b64encode(s=s)\n'\n '    value1 = base64.urlsafe_b64decode(s=value0)\n'\n '    assert s == value1, (s, value1)\n') == ('# This test code was written by the `hypothesis.extra.ghostwriter` module\n'\n '# and is provided under the Creative Commons Zero public domain dedication.\n'\n '\n'\n 'import base64\n'\n 'from hypothesis import given, strategies as st\n'\n '\n'\n '# TODO: replace st.nothing() with appropriate strategies\n'\n '\n'\n '\n'\n '@given(\n'\n '    adobe=st.booleans(),\n'\n '    b=st.nothing(),\n'\n '    foldspaces=st.booleans(),\n'\n '    ignorechars=st.just(b" \\t\\n\\r\\x0b"),\n'\n '    pad=st.booleans(),\n'\n '    wrapcol=st.just(0),\n'\n ')\n'\n 'def test_roundtrip_a85encode_a85decode(adobe, b, foldspaces, ignorechars, '\n 'pad, wrapcol):\n'\n '    value0 = base64.a85encode(\n'\n '        b=b, foldspaces=foldspaces, wrapcol=wrapcol, pad=pad, adobe=adobe\n'\n '    )\n'\n '    value1 = base64.a85decode(\n'\n '        b=value0, foldspaces=foldspaces, adobe=adobe, '\n 'ignorechars=ignorechars\n'\n '    )\n'\n '    assert b == value1, (b, value1)\n'\n '\n'\n '\n'\n '@given(casefold=st.booleans(), s=st.nothing())\n'\n 'def test_roundtrip_b16encode_b16decode(casefold, s):\n'\n '    value0 = base64.b16encode(s=s)\n'\n '    value1 = base64.b16decode(s=value0, casefold=casefold)\n'\n '    assert s == value1, (s, value1)\n'\n '\n'\n '\n'\n '@given(casefold=st.booleans(), map01=st.none(), s=st.nothing())\n'\n 'def test_roundtrip_b32encode_b32decode(casefold, map01, s):\n'\n '    value0 = base64.b32encode(s=s)\n'\n '    value1 = base64.b32decode(s=value0, casefold=casefold, map01=map01)\n'\n '    assert s == value1, (s, value1)\n'\n '\n'\n '\n'\n '@given(altchars=st.none(), s=st.nothing(), validate=st.booleans())\n'\n 'def test_roundtrip_b64encode_b64decode(altchars, s, validate):\n'\n '    value0 = base64.b64encode(s=s, altchars=altchars)\n'\n '    value1 = base64.b64decode(s=value0, altchars=altchars, '\n 'validate=validate)\n'\n '    assert s == value1, (s, value1)\n'\n '\n'\n '\n'\n '@given(b=st.nothing(), pad=st.booleans())\n'\n 'def test_roundtrip_b85encode_b85decode(b, pad):\n'\n '    value0 = base64.b85encode(b=b, pad=pad)\n'\n '    value1 = base64.b85decode(b=value0)\n'\n '    assert b == value1, (b, value1)\n'\n '\n'\n '\n'\n '@given(input=st.nothing(), output=st.nothing())\n'\n 'def test_roundtrip_encode_decode(input, output):\n'\n '    value0 = base64.encode(input=input, output=output)\n'\n '    value1 = base64.decode(input=value0, output=output)\n'\n '    assert input == value1, (input, value1)\n'\n '\n'\n '\n'\n '@given(s=st.nothing())\n'\n 'def test_roundtrip_encodebytes_decodebytes(s):\n'\n '    value0 = base64.encodebytes(s=s)\n'\n '    value1 = base64.decodebytes(s=value0)\n'\n '    assert s == value1, (s, value1)\n'\n '\n'\n '\n'\n '@given(s=st.nothing())\n'\n 'def test_roundtrip_standard_b64encode_standard_b64decode(s):\n'\n '    value0 = base64.standard_b64encode(s=s)\n'\n '    value1 = base64.standard_b64decode(s=value0)\n'\n '    assert s == value1, (s, value1)\n'\n '\n'\n '\n'\n '@given(s=st.nothing())\n'\n 'def test_roundtrip_urlsafe_b64encode_urlsafe_b64decode(s):\n'\n '    value0 = base64.urlsafe_b64encode(s=s)\n'\n '    value1 = base64.urlsafe_b64decode(s=value0)\n'\n '    assert s == value1, (s, value1)\n')
    # This test code was written by the `hypothesis.extra.ghostwriter` module
    # and is provided under the Creative Commons Zero public domain dedication.
    
    import base64
    from hypothesis import given, strategies as st
    
    # TODO: replace st.nothing() with appropriate strategies
    
    
    @given(
        adobe=st.booleans(),
        b=st.nothing(),
        foldspaces=st.booleans(),
        ignorechars=st.just(b" \t\n\r\x0b"),
        pad=st.booleans(),
        wrapcol=st.just(0),
    )
    def test_roundtrip_a85encode_a85decode(adobe, b, foldspaces, ignorechars, pad, wrapcol):
        value0 = base64.a85encode(
            b=b, foldspaces=foldspaces, wrapcol=wrapcol, pad=pad, adobe=adobe
        )
        value1 = base64.a85decode(
            b=value0, foldspaces=foldspaces, adobe=adobe, ignorechars=ignorechars
        )
        assert b == value1, (b, value1)
    
    
    @given(casefold=st.booleans(), s=st.nothing())
    def test_roundtrip_b16encode_b16decode(casefold, s):
        value0 = base64.b16encode(s=s)
        value1 = base64.b16decode(s=value0, casefold=casefold)
        assert s == value1, (s, value1)
    
    
    @given(casefold=st.booleans(), map01=st.none(), s=st.nothing())
    def test_roundtrip_b32encode_b32decode(casefold, map01, s):
        value0 = base64.b32encode(s=s)
        value1 = base64.b32decode(s=value0, casefold=casefold, map01=map01)
  +     assert s == value1, (s, value1)
  + 
  + 
  + @given(casefold=st.booleans(), s=st.nothing())
  + def test_roundtrip_b32hexencode_b32hexdecode(casefold, s):
  +     value0 = base64.b32hexencode(s=s)
  +     value1 = base64.b32hexdecode(s=value0, casefold=casefold)
        assert s == value1, (s, value1)
    
    
    @given(altchars=st.none(), s=st.nothing(), validate=st.booleans())
    def test_roundtrip_b64encode_b64decode(altchars, s, validate):
        value0 = base64.b64encode(s=s, altchars=altchars)
        value1 = base64.b64decode(s=value0, altchars=altchars, validate=validate)
        assert s == value1, (s, value1)
    
    
    @given(b=st.nothing(), pad=st.booleans())
    def test_roundtrip_b85encode_b85decode(b, pad):
        value0 = base64.b85encode(b=b, pad=pad)
        value1 = base64.b85decode(b=value0)
        assert b == value1, (b, value1)
    
    
    @given(input=st.nothing(), output=st.nothing())
    def test_roundtrip_encode_decode(input, output):
        value0 = base64.encode(input=input, output=output)
        value1 = base64.decode(input=value0, output=output)
        assert input == value1, (input, value1)
    
    
    @given(s=st.nothing())
    def test_roundtrip_encodebytes_decodebytes(s):
        value0 = base64.encodebytes(s=s)
        value1 = base64.decodebytes(s=value0)
        assert s == value1, (s, value1)
    
    
    @given(s=st.nothing())
    def test_roundtrip_standard_b64encode_standard_b64decode(s):
        value0 = base64.standard_b64encode(s=s)
        value1 = base64.standard_b64decode(s=value0)
        assert s == value1, (s, value1)
    
    
    @given(s=st.nothing())
    def test_roundtrip_urlsafe_b64encode_urlsafe_b64decode(s):
        value0 = base64.urlsafe_b64encode(s=s)
        value1 = base64.urlsafe_b64decode(s=value0)
        assert s == value1, (s, value1)
```

No idea how to debug this further.

**Comment by Zac-HD** [comment]
Pretty simple: [`base64.b32hexencode()` is new in Python 3.10](https://docs.python.org/3.10/library/base64.html#base64.b32hexencode), so we'll need to expect the new output on newer versions.

**Comment by Zac-HD** [comment]
Closed by 643b4fbb4874fa50efe72e1d3b1aaecc40de0e5f, which will land in `master` shortly.
